In [15]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import LabelEncoder
import multiprocessing as mp


In [16]:
df = pd.read_csv('C:/Users/yigit/OneDrive/Belgeler/GitHub/FlightPricePrediction/FlightPricePrediction.csv')

df

Unnamed: 0   airline   flight source_city departure_time stops  \
0                0  SpiceJet  SG-8709       Delhi        Evening  zero   
1                1  SpiceJet  SG-8157       Delhi  Early_Morning  zero   
2                2   AirAsia   I5-764       Delhi  Early_Morning  zero   
3                3   Vistara   UK-995       Delhi        Morning  zero   
4                4   Vistara   UK-963       Delhi        Morning  zero   
...            ...       ...      ...         ...            ...   ...   
300148      300148   Vistara   UK-822     Chennai        Morning   one   
300149      300149   Vistara   UK-826     Chennai      Afternoon   one   
300150      300150   Vistara   UK-832     Chennai  Early_Morning   one   
300151      300151   Vistara   UK-828     Chennai  Early_Morning   one   
300152      300152   Vistara   UK-822     Chennai        Morning   one   

         arrival_time destination_city     class  duration  days_left  price  
0               Night           Mumbai   Economy      2.17          1   5953  
1             Morning           Mumbai   Economy      2.33          1   5953  
2       Early_Morning           Mumbai   Economy      2.17          1   5956  
3           Afternoon           Mumbai   Economy      2.25          1   5955  
4             Morning           Mumbai   Economy      2.33          1   5955  
...               ...              ...       ...       ...        ...    ...  
300148        Evening        Hyderabad  Business     10.08         49  69265  
300149          Night        Hyderabad  Business     10.42         49  77105  
300150          Night        Hyderabad  Business     13.83         49  79099  
300151        Evening        Hyderabad  Business     10.00         49  81585  
300152        Evening        Hyderabad  Business     10.08         49  81585  

[300153 rows x 12 columns]

Unnamed: 0: Sayısal indeks veya satır numarası.

airline: Havayolu şirketi adı (örn. "SpiceJet" veya "AirAsia").

flight: Uçuş numarası veya kodu (örn. "SG-8709" veya "I5-764").

source_city: Kalkış şehri (örn. "Delhi").

departure_time: Kalkış zamanı (örn. "Evening" veya "Early_Morning").

stops: Uçuşta yapılan duraklama sayısı veya "zero" (duraksız uçuşlar için).

arrival_time: Varış zamanı (örn. "Night" veya "Morning").

destination_city: Varış şehri (örn. "Mumbai").

class: Uçuş sınıfı (örn. "Economy").

duration: Uçuş süresi (örn. 2.17 saat).

days_left: Uçuşa kaç gün kaldığı bilgisi (örn. 1 gün).

price: Uçuşun fiyatı (örn. 5953).

In [17]:
df.rename(columns={'Unnamed: 0': 'Flight Number'}, inplace=True)

In [21]:
print(df)

   Flight Number  Class  Price
0              0      1   5953
1              1      1   5953
2              2      1   5956
3              3      0   5955
4              4      0   5955


In [19]:
data = {
    'Flight Number': [0, 1, 2, 3, 4],
    'Class': ['Economy', 'Economy', 'Economy', 'Business', 'Business'],
    'Price': [5953, 5953, 5956, 5955, 5955]
}

df = pd.DataFrame(data)

# Business ve Economy uçuşları ayırma
business_flights = df[df['Class'] == 'Business']
economy_flights = df[df['Class'] == 'Economy']

print("Business Uçuşları:")
print(business_flights)

print("\nEconomy Uçuşları:")
print(economy_flights)

Business Uçuşları:
   Flight Number     Class  Price
3              3  Business   5955
4              4  Business   5955

Economy Uçuşları:
   Flight Number    Class  Price
0              0  Economy   5953
1              1  Economy   5953
2              2  Economy   5956


In [20]:
le = LabelEncoder()
df['Class'] = le.fit_transform(df['Class'])
print(df)


   Flight Number  Class  Price
0              0      1   5953
1              1      1   5953
2              2      1   5956
3              3      0   5955
4              4      0   5955
